drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+
+  finfo("CMD: %02x Address: %04lx addrlen=%d\n",
+        cmd, (unsigned long)addr, addrlen);
+
+  cmdinfo.flags   = QSPICMD_ADDRESS;
+  cmdinfo.addrlen = addrlen;
+  cmdinfo.cmd     = cmd;
+  cmdinfo.buflen  = 0;
+  cmdinfo.addr    = addr;
+  cmdinfo.buffer  = NULL;
+
+  return QSPI_COMMAND(priv->qspi, &cmdinfo);
+}
+
+/****************************************************************************
+ * Name: gd55_read_bytes
+ *
+ * Description:
+ *   Read data from the device
+ *
+ * Input Parameters:
+ *   priv         - a reference to the device structure
+ *   buffer       - pointer to buffer to read the data to
+ *   address      - address to read from
+ *   buflen       - number of bytes to read (buffer length)
+ *
+ * Returned Value:
+ *   Zero (OK) on SUCCESS, a negated errno on value of failure
+ *
+ ****************************************************************************/
+
+int gd55_read_bytes(FAR struct gd55_dev_s *priv, FAR uint8_t *buffer,
+                    off_t address, size_t buflen)
+{
+  bool                  mode_4byte_addr;
+  int                   ret;
+  struct qspi_meminfo_s meminfo;
+
+  /* Check if any address exceeds range of 3 byte addressing */
+
+  if ((address + buflen) >= MODE_3BYTE_LIMIT)
+    {
+      gd55_command(priv, GD55_EN4B);
+      mode_4byte_addr = true;
+    }
+
+  finfo("4byte mode: %s\taddress: %08lx\tnbytes: %d\n",
+         mode_4byte_addr ? "true" : "false", (long)address, (int)buflen);
+
+  meminfo.flags   = QSPIMEM_READ | QSPIMEM_QUADIO;
+  meminfo.dummies = GD55_QC_READ_DUMMIES;
+  meminfo.buflen  = buflen;
+  meminfo.cmd     = GD55_QC_READ;
+  meminfo.addr    = address;
+  meminfo.addrlen = mode_4byte_addr ? 4 : 3;
+  meminfo.buffer  = buffer;
+
+  ret = QSPI_MEMORY(priv->qspi, &meminfo);
+  if (mode_4byte_addr)
+    {
+      gd55_command(priv, GD55_DIS4B);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: gd55_write_page
+ *
+ * Description:
+ *   Write a page of data to the device
+ *
+ * Input Parameters:
+ *   priv         - a reference to the device structure
+ *   buffer       - pointer to the buffer with the data to write
+ *   address      - address to write to
+ *   buflen       - number of bytes to write (buffer length)
+ *
+ * Returned Value:
+ *   Zero (OK) on SUCCESS, a negated errno on value of failure
+ *
+ ****************************************************************************/
+
+int gd55_write_page(FAR struct gd55_dev_s *priv,
+                    FAR const uint8_t *buffer,
+                    off_t address, size_t buflen)
+{
+  struct qspi_meminfo_s meminfo;
+  uint8_t               status;
+  unsigned int          npages;
+  int                   ret;
+  int                   i;
+
+  npages = (buflen >> GD55_PAGE_SHIFT);
+
+  /* Check if address exceeds range of 3 byte addressing */
+
+  if ((address + buflen) >= MODE_3BYTE_LIMIT)
+    {
+    gd55_command(priv, GD55_EN4B);
```

Comment:
run checkpatch.sh

Reply:
I did before submitting PR and there are no errors? Why did it miss this one!!??

Will correct it and rerun checkpatch.sh as always.

Reply:
done